### PR TITLE
Update 40-fortigate_2_ecs

### PR DESCRIPTION
**Pull request by nicpenning**
I am seeing a lot of tcp/1234, TCP_3933, udp/1122, UDP/5533 as the data in the network protocol when I believe these should be collapsed down to network.transport : tcp or udp.

I am not sure TCP_3933 are really protocols but rather just ports?

This adjustment will ensure that network.protocol will contain: aol, rdp, ssh, telnet, etc.. by:

1. Lowercase network protocol for standardization and less regular expressions to match on in the next step.
2. If network.protocol starts with tcp or udp, add it to a new field called network.transport.
3. Remove the network.protocol field if the network.transport field exists. Out of all the data I have seen, network.protocol doesn't seem to exist, so this should work until someone can find a better way.

Let me know if this helps!

### CODE REVIEW

**Comment by enotspe** [comment]
According to ECS, `network.transport` is "_**Same as network.iana_number**, but instead using the Keyword name of the transport layer (udp, tcp, ipv6-icmp, etc.)_", so the most accurate way to populate it will be to do a lookup match from [network.iana_number](https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml). There a lot more values than just tcp or udp. Additionally, `network.transport` should always be populated because it is a `core` field.

`network.protocol` value is more arbitrary since it is "_L7 Network protocol name_". We should also take into consideration that SIEM UI uses `network.protocol` and not `network.transport` for its timeline´s renders, so even when it does not have a proper name (like tcp/1234), it might be useful to have it. I agree that this field needs to be lowercased to follow [ECS guidelines](https://www.elastic.co/guide/en/ecs/current/ecs-guidelines.html#_guidelines_for_field_names)

Also, consider that **TCP_1234** is a custom service that has been configured on the firewall, while **tcp/1234** is Fortinet´s default when no service matches proto/dstport combination.

What do you think?

**Comment by nicpenning** [comment]
Good information!

I think if the iana number exists in the log then we should definitely generate the network.transport field with the correlated keyword data. I think I remember seeing this data in the traffic log, but will have to revisit. 

As for the transport + port I don't see the value in calling that the protocol because we should be able to see port and transport in two different fields that already contain that data.

Where at in the SIEM app do you find this beneficial? I am thinking if the service doesn't exist we just look at alternative fields to understand the traffic. It doesn't seem like transport/port matches the description of the network protocol. For example tcp/22 is not really a protocol because I could serve a protocol such as http on that transport / port combination. In my mind, I am thinking that if no service/protocol exists I am not sure this a good place holder for that information. As an analyst this should prompt me to look at other sources and fields to understand what is happening in the traffic. Some analysts might see the "protocol" tcp/80 and assume it's HTTP when really it could be FTP for whatever reason. If I can take it a step further and assuming the networking device definitively knows that HTTP is occurring on that port then no guess work is to be had.

I can still be persuaded in other directions but just wanted to throw my perspective out there.

I think this is a great conversation!

So in short (TLDR):

- Map iana number to transport when possible, if not, use the transport from tcp/1234.
- Use network.protocol when the protocol is known, otherwise dont' fill it with transport/port combinations.

**Comment by enotspe** [comment]
fortigates deliver `proto`, `dstport` and `service` fields. `proto` and `service` are always delivered on traffic logs.

And they translate to ECS

`proto` --> `network.iana_number`
`dstport` --> `destination.port`
`service` --> `network.protocol`

So I don´t see any case where we should not get `network.transport` from no other than a lookup of `network.iana_number`. Maybe this will be a little more difficult to implement.

About `network.protocol`, I do agree that it should be taken only when the protocol is known. That makes lot of sense. 

Personally, I found very useful to always copy `service` to `network.protcol`, even when it is tcp/1234. It is like a free hash of proto/dstport and I can compare it to the other known protocols, like when you get HTTP on `service`.

What must be taken into account is that Fortinet populates `service` with the names that it has on its service object list. It does not perfom any check on the traffic for populating that field. So even if you are sending botnet traffic through udp/53, it will still be labeled as DNS. The field that has the output of the matched traffic pattern is `app`.

**Comment by enotspe** [comment]
with the current pipelines logic, [common_ecs](https://github.com/enotspe/fortinet-2-elasticsearch/blob/master/logstash/50-common_ecs) will be a better place to perform this enhacements.
